plugin/storage/badger/spanstore/read_write_test.go

```diff
@@ -25,7 +25,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
+	assert "github.com/stretchr/testify/require"
```

Comment:
All the tests use ``assert`` as keyword, but the way assert works is that it doesn't actually fail the test. This could cause some weird test errors that are not real (such as ``assert.NoError(err)`` not failing the test but going on).

Reply:
From what I can see, it is only the badger plugin that renames require to assert. Other parts of the code use both `assert` and `require`. Think it would be better to be explicit about using the `require` package.

Reply:
Yes, only badger renames, but a lot of tests in Jaeger would need the same rename. For example, consider test in https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/factory_test.go#L82 , if the f.initialize would throw an error (which is only asserted, not required), then what you will get is an error at the line 82 (a nil pointer exception), although the test should have stopped at line 77, since rest of the lines should not be executed (and can't be).

Same ``assert.NoError`` without actually checking if it threw an error is done ~300 times inside Jaeger codebase and many of them would cause fatal failures later in the code. But almost everyone thinks assert should stop the test (as nothing afterwards can usually be trusted). A simple rename allows to keep the defacto wording while making the behavior as expected. Even the require package talks about assertions instead of requirements: 

```
(Package require implements the same assertions as the `assert` package but stops test execution when a test fails. )
```

Reply:
Agree that this may be a problem - but not sure renaming is helping as it makes the code more ambigous - reviewer would need to look at the import to understand the actual behaviour. My personal preference would to ensure the code is clear, and identify the other problem areas and update them.

@yurishkuro thoughts?

Reply:
assert and require are semantically different and should be used explicitly as needed ("just alert" vs. "fail immediately"). Renaming feels like a bad practice that would only make the code harder to understand.

Reply:
also, `require` is an optimization over `assert`. In both cases the test will fail and print something, but if the error was fatal and we only alerted on it, then it could be obscured by a lot of output from unrelated failures, since the test should not have proceeded further.